docs/build-node-interaction.md

```diff
@@ -0,0 +1,163 @@
+```
+
+Some return values may not appear meaningful at first glance. Polkadot uses
+[SCALE encoding](https://substrate.dev/docs/en/next/conceptual/core/codec) as a format that is
+suitable for resource-constrained execution environments. You will need to decode the information
+and use the chain metadata (`state_getMetadata`) to obtain human-readable information.
+
+### Tracking the Chain Head
+
+Use `chain_subscribeFinalizedHeads` to subscribe to a stream of hashes of finalized headers, or
+`chain_FinalizedHeads` to fetch the latest hash of the finalized header. Use `chain_getBlock` to
+get the block associated with that hash. `chain_getBlock` only accepts block hashes, so if you need
+to query intermediate blocks, use `chain_getBlockHash` to get the block hash from a block number.
+
+## Substrate API Sidecar
+
+Parity maintains an RPC client that exposes a limited set of endpoints. It handles the metadata and
+codec logic so that you are always dealing with decoded information. It also aggregates information
+that an infrastructure business may need for accounting and auditing, e.g. transaction fees.
+
+The sidecar can fetch blocks, get the balance of an address atomically (i.e., with a corresponding
+block number), get the chain's metadata, and submit transactions to a node's transaction queue. If
+you have any feature/endpoint requests, log an issue in the
+[repo](https://github.com/paritytech/substrate-api-sidecar).
+
+The client runs on an HTTP host. The following examples use python3, but you can query any way you
+prefer at `http://HOST:PORT/`. The default is `http://127.0.0.1:8080`.
+
+### Fetching a Block
+
+Fetch a block using the `block/number` endpoint. To get the chain tip, omit the block number.
+
+```python
+import requests
+import json
+
+url = 'http://127.0.0.1:8080/block/1432511'
+response = requests.get(url)
+if response.ok:
+	block_info = json.loads(response.text)
+	print(block_info)
+```
+
+This returns a fully decoded block. In the `balances.transfer` extrinsic, the `partialFee` item is
+the transaction fee. It is called "partial fee" because the [total fee](build-protocol-info#fees)
+would include the `tip` field.
+
+```python
+{'number': 1432511,
+ 'hash': '0xd0807525ff44b9fe1185d88568d22bf405e5510c96f0f3e4c2168deacba8cd32',
+ 'parentHash': '0x668aab2ed0df7bf530f4af6a0e732f525ef648520650da981d349a3c5a56dacb',
+ 'stateRoot': '0x5a37e58467f65dc333cb21c8784f2b5891173b163322ac552d73c383355f436d',
+ 'extrinsicsRoot': '0x59e92902c90aeeec87cfdd076220b38f9ab77cc98a15456c7bbfb0de883be711',
+ 'logs': [{'type': 'PreRuntime',
+   'index': 6,
+   'value': [1161969986,
```

Comment:
It might be worth noting that there is no guarantee (at least AFAIU) that the numbers are going to be plain numbers. A lot of the values are internally bigints, but since JS number type is a float, with integer precision of 53 bits, anything larger than `Number.MAX_SAFE_INTEGER === 2**53-1` would have to be printed out as a string. Basically:

```
{
    "safe": 9007199254740991,
    "huge": "3783023686991216042",
}
```